tool/src/java/org/sakaiproject/gradebookng/tool/panels/StudentNameCellPanel.java

```diff
@@ -51,9 +51,13 @@ public void onClick(AjaxRequestTarget target) {
 				
 				GradebookPage gradebookPage = (GradebookPage) this.getPage();
 				final ModalWindow window = gradebookPage.getStudentGradeSummaryWindow();
-				
+
+				window.setOutputMarkupId(true);
+
```

Comment:
This is where I presume we can be clever about asking the window if it `isShown` and perform an `addOrReplace(Component...)` so the content changes in the already displaying modal.